### DISCUSSION TITLE
VSCode batch-replace script helps developers with new INTERFACE syntax

### DISCUSSION THREAD

**Discussion by RieksJ** [AmpersandTarski/Ampersand#1238] [Show and tell]
I'm not sure where else to put it, so basically this issue asks where we should put this stuff.

The 'stuff' is a (self documented) [vscode-batch-replace-script.txt](https://github.com/AmpersandTarski/Ampersand/files/5828173/vscode-batch-replace-script.txt) (note: updated with bugfixes!) that helps developers upgrade to the new interface syntax, by replacing the old interface syntax (e.g. `ROWS` or `COLSNL`) with the new syntax (`BOX <FORM>` resp. `BOX <TABLE noHeader>`) in all `.adl` and `.ifc` files that they have in an entire VSCode workspace. If you have additional extensions, you can edit the `filter` command on line 17 of the script accordingly.

The script does not guarantee that the semantics remain intact. After all, this is a breaking change in Ampersand.

You can run this script by the [Batch Replacer extension](https://marketplace.visualstudio.com/items?itemName=angelomollame.batch-replacer) of VSCode. After installation, do as if you are editing the script, press Ctrl-Shift-P (as you are editing the script), then search for `Batch Replacer`, and execute it.

**Comment by sjcjoosten** [comment]
This is cool, I often look for this kind of 'automatic conversion after a syntax change' stuff and I never know where to find it, so right place or not: thanks for putting it somewhere! I also wouldn't know where to put this, as the usefulness of this will slowly expire as people upgrade to the new syntax, and most places to put stuff are meant for the permanent putting of stuff, but since issues only remain open temporarily, this might be the best place we have for it. Too bad we don't have a "news" section on some website. Perhaps @stefjoosten has an idea on where to put this?

**Comment by hanjoosten** [comment]
Cool indeed. Depending on the usecase, your language, you might be interested in [tree-sitter](https://tree-sitter.github.io/tree-sitter/) as well.

**Comment by RieksJ** [comment]
Since you're so enthusiastic about this, I modified the original post to include my updated script, which has some bugfixes and will now modify both `*.adl` and `*.ifc` files in one run.

**Comment by stefjoosten** [comment]
Way to go! This issue relates to #1084
